### PR TITLE
Added primary sales integration in demo scene

### PR DESCRIPTION
**Pull request by andygruening**
Note that my branch is based on Tomas' QR Code branch, which has not been merged yet.

### CODE REVIEW

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Utils/UnityWebRequestExtensions.cs
Comment (BellringerQuinn) [inline Packages/Sequence-Unity/Sequence/SequenceSDK/Utils/UnityWebRequestExtensions.cs:23]: This doesn't seem to be used?

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs
Comment (BellringerQuinn) [inline Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs:19]: Can you add this contract code to the repo in the testchain/contracts directory?

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC721Sale.cs
Comment (BellringerQuinn) [inline Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC721Sale.cs:19]: Can you add this contract code to the repo in the testchain/contracts directory?

**Comment by BellringerQuinn** [comment]
Amazing work Andy!

File: Packages/Sequence-Unity/Sequence/Samples~/Setup/Resources/SequenceConfig.asset
Comment (BellringerQuinn) [reply Packages/Sequence-Unity/Sequence/Samples~/Setup/Resources/SequenceConfig.asset:32]: Ahh I see now, only if it is imported via Samples it is blocked by .gitignore. I'd recommend placing your config vars in that way instead. I don't think we want new integrators to spam one of your projects with requests as a "default" value

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySalePage.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySalePage.cs:65]: Yes you are right, removed!

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Utils/UnityWebRequestExtensions.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceSDK/Utils/UnityWebRequestExtensions.cs:23]: Done

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs:41]: I love it, but I changed it if its more readable

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/TransferFundsViaQR.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/TransferFundsViaQR.cs:16]: Done

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySaleTile.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySaleTile.cs:16]: It's the base class for TextMeshProUGUI and all TMP texts, but I prefer using that because it's much more readable.

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySaleStateERC1155.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySaleStateERC1155.cs:26]: Done

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySalePage.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySalePage.cs:49]: Done

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Prefabs/GameObject.prefab
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Prefabs/GameObject.prefab:1]: Done

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC721Sale.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC721Sale.cs:12]: Done

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySaleStateERC1155.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySaleStateERC1155.cs:98]: Good to know thanks!

File: Packages/Sequence-Unity/Sequence/Samples~/Setup/Resources/SequenceConfig.asset
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/Samples~/Setup/Resources/SequenceConfig.asset:32]: Yes I see, done!

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC721Sale.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC721Sale.cs:19]: Same as above.

File: Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySalePage.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/PrimarySales/PrimarySalePage.cs]: I added a serializable GenericObjectPool object, its only used for Primary Sales right now. I think we can switch to that class for the other integrations and then remove the regular 'ObjectPool' class. What do you think?

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs
Comment (BellringerQuinn) [reply Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs:19]: Thanks 😄 

No need to test it really. Just useful to have it there so that readers and maintainers can find the contract source code easily should they ever need it. Ideally, they can be compiled with hardhat in case they need to be modified for whatever reason, though since these aren't used for testing and are our templated contracts, it should be fine.

The testchain is used primarily via the Makefile and is useful for testing against a local chain (ideal for ethereum library work).

**Review by BellringerQuinn — APPROVED** [review]
Looks great Andy! Thanks for the tweaks

File: Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs
Comment (andygruening) [reply Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/Contract/ERC1155Sale.cs:19]: Awesome now I understand, thanks for the explanation Quinn!

**Pull request merged**
